builders/ethereum/dev-env/hardhat.md

```diff
@@ -83,7 +99,55 @@ Next, you'll need to modify your configuration file to add the network configura
 - `chainId` - the chain ID, which is used to validate the network
 - `accounts` - the accounts that can be used to deploy and interact with contracts. You can either enter an array of the private keys for your accounts or use an [HD Wallet](https://github.com/ethereumbook/ethereumbook/blob/develop/05wallets.asciidoc#hierarchical-deterministic-wallets-bip-32bip-44){target=\_blank}
 
-For this example, the network will be Moonbase Alpha, but you can modify the configuration to use any of the Moonbeam networks:
+Hardhat 3 includes an encrypted secrets manager that makes it easier to handle sensitive information like private keys. This ensures you don't have to hardcode secrets in your source code or store them in plain text.
+
+!!! note
+    The encrypted secrets manager is only available in Hardhat 3 or higher. As of writing this guide, Hardhat 3 is in alpha. You can install the latest alpha version with:
+
+    ```bash
+    npm install hardhat@3.0.0-next.5
+    ```
+
+    For the latest releases and updates, check the [Hardhat releases page](https://github.com/NomicFoundation/hardhat/releases/).
+
+To use encrypted secrets, you'll need to:
+
+1. Install Hardhat 3 or later:
+```bash
+npm install --save-dev hardhat@^3.0.0
+```
+
+2. Set up your secrets using the keystore:
+
+=== "Moonbeam"
+
+    ```bash
```

Comment:
When I got to this step and ran `npx hardhat keystore set MOONBASE_RPC_URL`, that's the first instance I got anything about ESM projects. It came back with:
dawnkelly@Mac hardhat % npx hardhat keystore set MOONBASE_RPC_URL
Hardhat only supports ESM projects.

Please make sure you have `"type": "module"` in your package.json.

You can set it automatically by running:

npm pkg set type="module"

Reply:
With the latest version you'll get a prompt at initialization like: ` Hardhat only supports ESM projects. Would you like to change "package.json" to turn your project into ESM? (Y/n) · true` 

It'll work successfully after that.